### PR TITLE
Add support for running local containers

### PR DESCRIPTION
**Pull request by davidfowl**
- This support builds the application locally then volume mounts the application into the container when the --docker flag is specified. Added support for volume mounts and working directory to the DockerRunner.

Here's what mult-project looks like with `tye run --docker`

image

### CODE REVIEW

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        public async Task StartAsync(Model.Application application)
+        {
+            // This transforms a ProjectRunInfo into
+
+            foreach (var s in application.Services.Values)
+            {
+                if (s.Description.RunInfo is ProjectRunInfo project)
+                {
+                    await TransformProjectToContainer(application, s, project);
```
Comment (davidfowl) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: This builds the projects and can be done in parallel (like the process runner currently does)

File: src/Microsoft.Tye.Hosting/DockerRunner.cs [hunk 2]
```diff
@@ -68,6 +68,8 @@ private async Task StartContainerAsync(Tye.Hosting.Model.Application application
 
             var serviceDescription = service.Description;
             var environmentArguments = "";
+            var volumes = "";
+            var workingDirectory = docker.WorkingDirectory == null ? "" : $"-w {docker.WorkingDirectory}";
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/DockerRunner.cs]: nit: invert condition

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
+﻿using System;
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: nit: copyright

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs (hunk 1 above)
Comment (jkotalik) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: But it isn't in parallel today, correct? Why can't you do a Task.Run on each and await all of them at the end?

Comment (davidfowl) [reply src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: I copied what the other processors do. Not quite Task.Run but it goes parallel if the build finishes asynchronously.

**Pull request merged**